contents/javascript/Javascript-framework-Redux.md

```diff
@@ -0,0 +1,231 @@
+
+In general, it would be only apt to use Redux is when:
+
+1. You have reasonable amount of data changing over time
+1. You need a single source of truth
+1. You find keeping everything in a top-level parent component is not enough
+
+Another good rule of thumb is to figure out if you have any problems managing state in your application without Redux. If you don't have any, then you don't really need Redux.  
+
+As Dan Abramov, one of the creators of Redux, says with regards to [React]({{baseUrl}}/contents/javascript/Javascript-framework-React.html), a view library very commonly used with Redux:
+
+<box seamless>
+I would like to amend this: don't use Redux until you have problems with vanilla React.
+</box>
+
+There are many other sites and articles speaking of why you shouldn't or when you shouldn't use Redux. If you are interested, the following resources in this [section](({{baseUrl}}/contents/javascript/javascript-framework-redux#more-articles-on-when-to-use-redux)) provide more information and thoughts on when Redux is meant to be used.
+
+## Getting Started with Redux
+
+The [official Redux website]((https://redux.js.org/)) is a great place to get started. It includes:
+
+- [Installation instructions](https://redux.js.org/introduction/getting-started/)
+- A suggested list of resources encompassing [Basic](https://redux.js.org/introduction/getting-started/#just-the-basics) and [Intermediate](https://redux.js.org/introduction/getting-started/#intermediate-concepts) concepts.
+- A [basic](https://redux.js.org/basics/basic-tutorial) and [advanced](https://redux.js.org/advanced/advanced-tutorial) tutorial.
+
+This article is based on the the sections [Core Concepts](https://redux.js.org/introduction/core-concepts) and [Three Principles](https://redux.js.org/introduction/three-principles) of the introduction, so you may find the sections familiar and be able skim through those.
+
+Installing the [Redux DevTools](https://github.com/reduxjs/redux-devtools) would greatly help in terms of seeing Redux in action with whichever project or tutorial you are working on.
+
+Redux especially works well with React, read up on [React Redux](https://react-redux.js.org/), which are official React bindings for Redux
+
+### More Articles on When to Use Redux 
+
+- [You Might Not Need Redux](https://medium.com/@dan_abramov/you-might-not-need-redux-be46360cf367)
+- [When should I use Redux?](https://redux.js.org/faq/general#when-should-i-use-redux)
+
+## Other Frontend Framework Specific State Management Libraries
+
+For the other popular frontend frameworks, they may have their own state management libraries more suited for them. State management libraries are generally inspired from one another, and you may find them follow similar general concepts as Redux. What is more important that each of these libraries are usually tailored to a specific front end library. The other frontend libraries and their corresponding state management libraries are:
```

Comment:
`Other popular ...`
`Many of the concepts of state management libraries are borrowed from..., hence ...`
Delete `What is more important...`
`Some other frontend libraries...`